pysqldb3/pysqldb3.py

```diff
@@ -2073,3 +2074,216 @@ def feature_class_to_table(self, path, table, schema=None, shp_name=None, gdal_d
 
         if temp:
             self.__run_table_logging([schema + "." + table], days=days)
+
+
+    def query_to_gpkg(self, query, path=None, gpkg_name=None, cmd=None, tbl_name = None, gdal_data_loc=GDAL_DATA_LOC,
+                     print_cmd=False, srid=2263):
+        """
+        Exports query results to a geopackage (.gpkg) file.
+        :param query: SQL query as string type
+        :param path: folder path for output gpkg
+        :param gpkg_name: filename for shape (should end in .gpkg)
+        :param cmd: GDAL command to overwrite default
+        :param tbl_name: Name of table in the Geopackage output
+        :param gdal_data_loc: Path to gdal data, if not stored in system env correctly
+        :param print_cmd: boolean to print ogr command (without password)
+        :param srid: SRID to manually set output to; defaults to 2263
+        :return:
+        """
+        # Temporarily sets temp flag to True
+        original_temp_flag = self.allow_temp_tables
+        self.allow_temp_tables = True
+
+        # Makes a temp table name
+        tmp_table_name = "tmp_query_to_gpkg_{}_{}".format(self.user,
+                                                         str(datetime.datetime.now())[:16].replace('-', '_').replace(
+                                                             ' ', '_').replace(':', ''))
+
+        # Create temp table to get column types
+        try:
+            # Drop the temp table
+            if self.type == PG:
+                self.query("drop table {}".format(tmp_table_name), internal=True, strict=False)
+            elif self.type == MS:
+                self.query("drop table #{}".format(tmp_table_name), internal=True, strict=False)
+        except Exception as e:
+            print(e)
+            pass
+
+        if self.type == PG:
+            self.query(f"""    
+            create temp table {tmp_table_name} as     
+            select * 
+            from ({query}) q 
+            limit 10
```

Comment:
How I understand what's happening (since this was copied from the `shp` function) is that it takes only 10 rows just to make the temp table small, and then it's using that temp table to parse out date/time columns so that it can reformat them before it runs the entire query

Reply:
But as an aside, Geopackage lets you name columns however long you want. This is different from shapefiles, which has a character limit

Reply:
does geopackage allow datetime format? if so this is not needed we do it for shp since that datatype is not an option